File: Data Structures & Algorithms/non-cyclical-number/submission-2.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n != 1:
            tmp = 0
            while n >= 1:
                tmp += (n % 10) ** 2
                n //= 10
            if tmp in seen:
                return False
            seen.add(tmp)
            n = tmp
        return True

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> seen;
        for (int num: nums) {
            if (!seen.contains(num)) {
                seen.emplace(num);
            }
            else {
                return true;
            }
        }
        return false;
    }
};